src/pages/taskQueue/TaskQueueTable/index.tsx

```diff
@@ -0,0 +1,168 @@
+import { useEffect, useMemo, useRef, useState } from "react";
+import styled from "@emotion/styled";
+import Badge from "@leafygreen-ui/badge";
+import { LGColumnDef, useLeafyGreenTable } from "@leafygreen-ui/table";
+import { Body, Disclaimer } from "@leafygreen-ui/typography";
+import { useTaskQueueAnalytics } from "analytics";
+import { StyledRouterLink, WordBreak } from "components/styles";
+import { BaseTable } from "components/Table/BaseTable";
+import { TablePlaceholder } from "components/Table/TablePlaceholder";
+import {
+  getVersionRoute,
+  getTaskRoute,
+  getUserPatchesRoute,
+} from "constants/routes";
+import { TaskQueueItem, TaskQueueItemType } from "gql/generated/types";
+import { formatZeroIndexForDisplay } from "utils/numbers";
+import { msToDuration } from "utils/string";
+
+type TaskQueueColumnData = Omit<TaskQueueItem, "revision">;
+interface TaskQueueTableProps {
+  taskQueue: TaskQueueColumnData[];
+  taskId?: string;
+}
+const estimateSize = () => 65;
+
+const TaskQueueTable: React.FC<TaskQueueTableProps> = ({
+  taskId,
+  taskQueue = [],
+}) => {
+  const taskQueueAnalytics = useTaskQueueAnalytics();
+  const tableContainerRef = useRef<HTMLDivElement>(null);
+  const [selectedRowIndexes, setSelectedRowIndexes] = useState<number[]>([]);
+  const columns = useMemo(
+    () => taskQueueTableColumns(taskQueueAnalytics.sendEvent),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [],
+  );
+
+  const table = useLeafyGreenTable<TaskQueueColumnData>({
+    data: taskQueue,
+    columns,
+    containerRef: tableContainerRef,
+    useVirtualScrolling: true,
+    virtualizerOptions: {
+      estimateSize,
+    },
+  });
+  const performedInitialScroll = useRef(false);
+  useEffect(() => {
+    if (taskId !== undefined && !performedInitialScroll.current) {
+      const i = taskQueue.findIndex((t) => t.id === taskId);
+      setSelectedRowIndexes([i]);
+      table.scrollToIndex(i, { align: "start" });
+
+      setTimeout(() => {
+        performedInitialScroll.current = true;
+        table.scrollToIndex(i, { align: "start" });
+      }, 200);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [taskId, taskQueue]);
+
+  return (
+    <StyledBaseTable
+      data-cy="task-queue-table"
+      table={table}
+      shouldAlternateRowColor
+      emptyComponent={<TablePlaceholder message="No tasks found in queue." />}
+      ref={tableContainerRef}
+      selectedRowIndexes={selectedRowIndexes}
+    />
+  );
+};
+
+const StyledBaseTable = styled(BaseTable)`
+  flex-grow: 1;
+`;
+
+const TaskCell = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+const taskQueueTableColumns = (
```

Comment:
it looks like the search icons in these column headers don't have any actual functionality, so they could be removed